src/libs/fsdrv/lv_fs_littlefs.c

```diff
@@ -0,0 +1,188 @@
+#include "../../../lvgl.h"
+#if LV_USE_FS_LITTLEFS
+
+#include "lfs.h"
+#include "../../core/lv_global.h"
+
+struct LittleFile {
+    lfs_file_t file;
+};
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static void * fs_open(lv_fs_drv_t * drv, const char * path, lv_fs_mode_t mode);
+static lv_fs_res_t fs_close(lv_fs_drv_t * drv, void * file_p);
+static lv_fs_res_t fs_read(lv_fs_drv_t * drv, void * file_p, void * buf, uint32_t btr, uint32_t * br);
+static lv_fs_res_t fs_write(lv_fs_drv_t * drv, void * file_p, const void * buf, uint32_t btw, uint32_t * bw);
+static lv_fs_res_t fs_seek(lv_fs_drv_t * drv, void * file_p, uint32_t pos, lv_fs_whence_t whence);
+static lv_fs_res_t fs_tell(lv_fs_drv_t * drv, void * file_p, uint32_t * pos_p);
+
+void lv_littlefs_set_handler(lfs_t * lfs)
+{
+    lv_fs_drv_t * drv = lv_fs_get_drv(LV_FS_LITTLEFS_LETTER);
+    drv->user_data = lfs;
+}
+
+/**
+ * Register a driver for the LittleFS File System interface
+ */
+void lv_fs_littlefs_init(void)
+{
+    lv_fs_drv_t * fs_drv_p = &(LV_GLOBAL_DEFAULT()->littlefs_fs_drv);
+    lv_fs_drv_init(&fs_drv);
+
+    fs_drv.letter = LV_FS_LITTLEFS_LETTER;
+    fs_drv.open_cb = fs_open;
+    fs_drv.close_cb = fs_close;
+    fs_drv.read_cb = fs_read;
+    fs_drv.write_cb = fs_write;
+    fs_drv.seek_cb = fs_seek;
+    fs_drv.tell_cb = fs_tell;
+
+    fs_drv.dir_close_cb = NULL;
+    fs_drv.dir_open_cb = NULL;
+    fs_drv.dir_read_cb = NULL;
+
+    lv_fs_drv_register(&fs_drv);
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+/**
+ * Open a file
+ * @param drv       pointer to a driver where this function belongs
+ * @param path      path to the file beginning with the driver letter (e.g. S:/folder/file.txt)
+ * @param mode      read: FS_MODE_RD, write: FS_MODE_WR, both: FS_MODE_RD | FS_MODE_WR
+ * @return          a file descriptor or NULL on error
+ */
+static void * fs_open(lv_fs_drv_t * drv, const char * path, lv_fs_mode_t mode)
+{
+    LV_UNUSED(drv);
+
+    int flags;
+    if(mode == LV_FS_MODE_WR)
+        flags = LFS_O_WRONLY;
+    else if(mode == LV_FS_MODE_RD)
+        flags = LFS_O_RDONLY;
+    else if(mode == (LV_FS_MODE_WR | LV_FS_MODE_RD))
+        flags = LFS_O_RDWR;
+
+    LittleFile * lf = lv_malloc(sizeof(*lf));
+    LV_ASSERT_NULL(lf);
+
+    lfs_t * lfs = drv->user_data;
+    int err = lfs_file_open(&lfs, &lf->file, path, flags);
+    if(err) {
+        return NULL;
+    }
+
+    return (void *)lf;
+}
+
+/**
+ * Close an opened file
+ * @param drv       pointer to a driver where this function belongs
+ * @param file_p    pointer to a file_t variable. (opened with fs_open)
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_close(lv_fs_drv_t * drv, void * file_p)
+{
+    LV_UNUSED(drv);
+    LittleFile * lf = (LittleFile *)file_p;
+
+    lfs_t * lfs = drv->user_data;
+    lfs_file_close(&lfs, &lf->file);
```

Comment:
Is it not a pointer variable that is defined? Why use the fetch address symbol? A lot of the pr is used in this way.